qiskit/circuit/library/arithmetic/exact_reciprocal.py

```diff
@@ -0,0 +1,53 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2019, 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""Exact reciprocal rotation."""
+
+from math import isclose
+import numpy as np
+from qiskit.circuit import QuantumCircuit, QuantumRegister
+from qiskit.extensions.quantum_initializer import UCRYGate
+
+
+class ExactReciprocal(QuantumCircuit):
+    r"""Exact reciprocal
+    .. math::
```

Comment:
Can you add some more docs here and also say that it is a part of HHL? 🙂 

```suggestion
    r"""Exact reciprocal
    
    .. math::
```